mantidproject/mantid
# Fix clang-tidy warning: misc-incorrect-roundings

Description of work.

This removes the warnings found by clang-tidy's [misc-incorrect-rounding](http://clang.llvm.org/extra/clang-tidy/checks/misc-incorrect-roundings.html). Where possible, I tried to remove `static_cast<int>`.

**To test:**



Please check that the behavior hasn't changed.

[std::round & std::lround documentation](http://www.cplusplus.com/reference/cmath/lround/)

[boost::math::iround documentation](http://www.boost.org/doc/libs/1_60_0/libs/math/doc/html/math_toolkit/rounding/round.html)

[warnings on the clang-tidy build](http://builds.mantidproject.org/view/Static%20Analysis/job/clang_tidy/115/warnings10Result/category.-1521249066/)

This PR is not associated with an issue.

Fortunately, this didn't change any test results and does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
